File: JAVA_SIS104/app/src/main/java/com/example/prueba_java/CalculadoraActivity.java
package com.example.prueba_java;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.prueba_java.backend.operaciones;

public class CalculadoraActivity extends AppCompatActivity {
    EditText edidtxtA, edidtxtB;
    Button buttonSuma, buttonResta, buttonMultiplicacion, buttonDivision, buttonSalirCalc,
            num1, num2, num3, num4, num5, num6, num7, num8, num9, num0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_calculadora);

        edidtxtA = findViewById(R.id.edidTextA);
        edidtxtB = findViewById(R.id.edidTextB);
        buttonSuma = findViewById(R.id.buttonS);
        buttonResta = findViewById(R.id.buttonR);
        buttonMultiplicacion = findViewById(R.id.buttonM);
        buttonDivision = findViewById(R.id.buttonD);
        buttonSalirCalc = findViewById(R.id.buttonSalirCalc);
        num1 = findViewById(R.id.buttonNum1);
        num2 = findViewById(R.id.buttonNum2);
        num3 = findViewById(R.id.buttonNum3);
        num4 = findViewById(R.id.buttonNum4);
        num5 = findViewById(R.id.buttonNum5);
        num6 = findViewById(R.id.buttonNum6);
        num7 = findViewById(R.id.buttonNum7);
        num8 = findViewById(R.id.buttonNum8);
        num9 = findViewById(R.id.buttonNum9);
        num0 = findViewById(R.id.buttonNum0);


        buttonSuma.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                /*int x = Integer.parseInt(edidtxtA.getText().toString());
                int y = Integer.parseInt(edidtxtB.getText().toString());
                operaciones Operaciones = new operaciones(x, y);
                Toast.makeText(CalculadoraActivity.this,"La suma es: " + Operaciones.suma(),Toast.LENGTH_LONG).show();*/
                String text = buttonSuma.getText().toString();
                edidtxtA.append(text);
            }
        });

        buttonResta.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int x = Integer.parseInt(edidtxtA.getText().toString());
                int y = Integer.parseInt(edidtxtB.getText().toString());
                operaciones Operaciones = new operaciones(x, y);
                Toast.makeText(CalculadoraActivity.this,"La resta es: " + Operaciones.resta(),Toast.LENGTH_LONG).show();
            }
        });

        buttonMultiplicacion.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int x = Integer.parseInt(edidtxtA.getText().toString());
                int y = Integer.parseInt(edidtxtB.getText().toString());
                operaciones Operaciones = new operaciones(x, y);
                Toast.makeText(CalculadoraActivity.this,"La multiplicacion es: " + Operaciones.multiplicacion(),Toast.LENGTH_LONG).show();
            }
        });

        buttonDivision.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //int x = Integer.parseInt(edidtxtA.getText().toString());
                //int y = Integer.parseInt(edidtxtB.getText().toString());
                operaciones Operaciones = new operaciones(Integer.parseInt(edidtxtA.getText().toString()), Integer.parseInt(edidtxtB.getText().toString()));
                Toast.makeText(CalculadoraActivity.this,"La division es: " + Operaciones.division(),Toast.LENGTH_LONG).show();
            }
        });

        buttonSalirCalc.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });

        num1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = num1.getText().toString();
                edidtxtA.append(text);
            }
        });

        num2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = num2.getText().toString();
                edidtxtA.append(text);
            }
        });

        num3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = num3.getText().toString();
                edidtxtA.append(text);
            }
        });

        num4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = num4.getText().toString();
                edidtxtA.append(text);
            }
        });

        num5.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = num5.getText().toString();
                edidtxtA.append(text);
            }
        });

        num6.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = num6.getText().toString();
                edidtxtA.append(text);
            }
        });

        num7.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = num7.getText().toString();
                edidtxtA.append(text);
            }
        });

        num8.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = num8.getText().toString();
                edidtxtA.append(text);
            }
        });

        num9.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = num9.getText().toString();
                edidtxtA.append(text);
            }
        });

        num0.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = num0.getText().toString();
                edidtxtA.append(text);
            }
        });


    }


}

File: KOTLIN_SIS104/app/src/main/java/com/example/prueba_kotlin/CalculadoraActivity.kt
package com.example.prueba_kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import com.example.prueba_kotlin.backend.operaciones

class CalculadoraActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_calculadora)

        val editTextA = findViewById<EditText>(R.id.EditTextA)
        val editTextB = findViewById<EditText>(R.id.EditTextB)
        val butSuma = findViewById<Button>(R.id.buttonSuma)
        val butResta = findViewById<Button>(R.id.buttonResta)
        val butMultiplicacion = findViewById<Button>(R.id.buttonMultiplicacion)
        val butDivision = findViewById<Button>(R.id.buttonDivision)
        val butIgual = findViewById<Button>(R.id.buttonIgual)
        val buttonsalircal = findViewById<Button>(R.id.buttonSalirCalc)
        val butDel = findViewById<Button>(R.id.buttonDel)
        val num0 = findViewById<Button>(R.id.buttonNum0)
        val num1 = findViewById<Button>(R.id.buttonNum1)
        val num2 = findViewById<Button>(R.id.buttonNum2)
        val num3 = findViewById<Button>(R.id.buttonNum3)
        val num4 = findViewById<Button>(R.id.buttonNum4)
        val num5 = findViewById<Button>(R.id.buttonNum5)
        val num6 = findViewById<Button>(R.id.buttonNum6)
        val num7 = findViewById<Button>(R.id.buttonNum7)
        val num8 = findViewById<Button>(R.id.buttonNum8)
        val num9 = findViewById<Button>(R.id.buttonNum9)
        val memoria = ArrayList<Double>()

        num0.setOnClickListener {
            val text = num0.text.toString()
            if(editTextA.equals("")){
                editTextA.append(text)
            }else{
                editTextA.append(memoria[0].toString())
            }
        }

        num1.setOnClickListener {
            val text = num1.text.toString()
            editTextA.append(text)
        }

        num2.setOnClickListener {
            val text = num2.text.toString()
            editTextA.append(text)
        }

        num3.setOnClickListener {
            val text = num3.text.toString()
            editTextA.append(text)
        }

        num4.setOnClickListener {
            val text = num4.text.toString()
            editTextA.append(text)
        }

        num5.setOnClickListener {
            val text = num5.text.toString()
            editTextA.append(text)
        }

        num6.setOnClickListener {
            val text = num6.text.toString()
            editTextA.append(text)
        }

        num7.setOnClickListener {
            val text = num7.text.toString()
            editTextA.append(text)
        }

        num8.setOnClickListener {
            val text = num8.text.toString()
            editTextA.append(text)
        }

        num9.setOnClickListener {
            val text = num9.text.toString()
            editTextA.append(text)
        }

        butSuma.setOnClickListener {
            val op = butSuma.text.toString()
            if(editTextB.text.toString().equals("Resultado")){
                editTextA.append(op)
            }else{
                editTextA.text.clear()
                editTextA.append(editTextB.text.toString() + op)
                editTextB.setText("Resultado")
            }
        }

        butResta.setOnClickListener {
            val op = butResta.text.toString()
            if(editTextB.text.toString().equals("Resultado")){
                editTextA.append(op)
            }else{
                editTextA.text.clear()
                editTextA.append(editTextB.text.toString() + op)
                editTextB.setText("Resultado")
            }
        }

        butMultiplicacion.setOnClickListener {
            val op = butMultiplicacion.text.toString()
            editTextA.append(op)
            if(editTextB.text.toString().equals("Resultado")){
                editTextA.append(op)
            }else{
                editTextA.text.clear()
                editTextA.append(editTextB.text.toString() + op)
                editTextB.setText("Resultado")
            }
        }

        butDivision.setOnClickListener {
            val op = butDivision.text.toString()
            editTextA.append(op)
            if(editTextB.text.toString().equals("Resultado")){
                editTextA.append(op)
            }else{
                editTextA.text.clear()
                editTextA.append(editTextB.text.toString() + op)
                editTextB.setText("Resultado")
            }
        }

        butIgual.setOnClickListener {
            val textOp = editTextA.text.toString()
            editTextB.text.clear()
            if(textOp.contains('+')){
                val op = textOp.split("+")
                val Operaciones = operaciones(op[0].toDouble(), op[1].toDouble())
                val result = Operaciones.suma()
                editTextB.setText(result.toString())
            }
            if(textOp.contains('-')){
                val op = textOp.split("-")
                val Operaciones = operaciones(op[0].toDouble(), op[1].toDouble())
                val result = Operaciones.resta()
                editTextB.setText(result.toString())
            }
            if(textOp.contains('*')){
                val op = textOp.split("*")
                val Operaciones = operaciones(op[0].toDouble(), op[1].toDouble())
                val result = Operaciones.multiplicacion()
                editTextB.setText(result.toString())
            }
            if(textOp.contains('/')){
                val op = textOp.split("/")
                val Operaciones = operaciones(op[0].toDouble(), op[1].toDouble())
                val result = Operaciones.division()
                editTextB.setText(result)
            }
        }

        butDel.setOnClickListener {
            editTextA.text.clear()
            editTextB.text.clear()
            editTextB.setText("Resultado")
        }

        buttonsalircal.setOnClickListener{
            finish()
        }
    }
}